lib/src/domain/usecase/implementation/up_coming_movies_usecase.dart

```diff
@@ -0,0 +1,15 @@
+import '../../../core/util/either_string_t.dart';
+import '../../entity/movie_entity.dart';
+import '../../repository/movie_repository_i.dart';
+import '../list_usecase_i.dart';
+
+class UpComingMoviesUseCase implements ListUseCaseI<List<Movie>> {
```

Comment:
Same as i mentioned above, you are creating for separate use case that are exactly the same, the only thing that you change is the repository call, you can keep it like this way, but maybe you can only have one use case, that centralized the 4 repository calls

Reply:
fixed